segment-event-notification/functions/track-sms.js

```diff
@@ -0,0 +1,28 @@
+exports.handler = async function(context, event, callback) {
```

Comment:
This is still not protected. I think it's tricky since the URL is guessable and would let you free form change the content.

Reply:
@dkundel i think we need a separate effort around this since it impacts more than just the Segment app. i would say lets not block this app on this and have a separate ticket in the next sprint to address it. thoughts?

Reply:
Doesn't the nature of segment initiating the incoming request require it be a publicly addressable route?

I'm not sure we even have a way of providing protection here

Reply:
I think there might be a protection mechanism that's possible, but I agree with @bharatmurali that implementing it as a separate effort makes the most sense. If we do this as a separate ticket, it'll allow for a bit more time researching how best to design that protection, and any mechanism that's designed can be implemented across the apps in the repo.

Reply:
Yeah I didn't mean protected as in `protected` Function but more as in adding a query param with some sort of random token. This would be properly solved once Functions supports access to headers. I have an idea for an intermediary solution that we can implement in a follow-on PR as long as we make sure we have it planned :)